model/vm_host.rb

```diff
@@ -243,16 +244,70 @@ def reset
     Hosting::Apis.reset_server(self)
   end
 
+  def check_storage_smartctl(ssh_session, devices)
+    devices.map { |device_name| ssh_session.exec!("sudo smartctl -j -H /dev/#{device_name} | jq .smart_status.passed").strip == "true" }.all?(true)
+  end
+
+  def check_storage_nvme(ssh_session, devices)
+    devices.map { |device_name| ssh_session.exec!("sudo nvme smart-log /dev/#{device_name} | grep \"critical_warning\" | awk '{print $3}'").strip == "0" }.all?(true)
+  end
+
+  def check_storage_read_write(ssh_session, devices)
+    lsblk_json_info = ssh_session.exec!("lsblk --json")
+    device_mount_points = devices.map { |device| SystemParser.get_device_mount_points_from_lsblk(lsblk_json_info, device) }
+
+    all_mount_points = []
+    device_mount_points.each do |device_mount_point|
```

Comment:
nit: variable naming is a bit confusing in this block. Maybe change it to something like this:

```ruby
    devices_with_mount_points.each do |device_mount_points|
      device_mount_points.each do |device_name, mount_points|
        mount_points.each do |mount_point|
          all_mount_points.append(mount_point)
        end
      end
```